google/cloud/bigtable/data/_sync_autogen/_mutate_rows.py

```diff
@@ -0,0 +1,182 @@
+        self.mutations = [_EntryWithProto(m, m._to_pb()) for m in mutation_entries]
+        self.remaining_indices = list(range(len(self.mutations)))
+        self.errors: dict[int, list[Exception]] = {}
+
+    def start(self):
+        """Start the operation, and run until completion
+
+        Raises:
+            MutationsExceptionGroup: if any mutations failed"""
+        try:
+            self._operation()
+        except Exception as exc:
+            incomplete_indices = self.remaining_indices.copy()
+            for idx in incomplete_indices:
+                self._handle_entry_error(idx, exc)
+        finally:
+            all_errors: list[Exception] = []
+            for idx, exc_list in self.errors.items():
+                if len(exc_list) == 0:
+                    raise core_exceptions.ClientError(
+                        f"Mutation {idx} failed with no associated errors"
+                    )
+                elif len(exc_list) == 1:
+                    cause_exc = exc_list[0]
+                else:
+                    cause_exc = bt_exceptions.RetryExceptionGroup(exc_list)
+                entry = self.mutations[idx].entry
+                all_errors.append(
+                    bt_exceptions.FailedMutationEntryError(idx, entry, cause_exc)
+                )
+            if all_errors:
+                raise bt_exceptions.MutationsExceptionGroup(
+                    all_errors, len(self.mutations)
+                )
+
+    def _run_attempt(self):
+        """Run a single attempt of the mutate_rows rpc.
+
+        Raises:
+            _MutateRowsIncomplete: if there are failed mutations eligible for
+                retry after the attempt is complete
+            GoogleAPICallError: if the gapic rpc fails"""
+        request_entries = [self.mutations[idx].proto for idx in self.remaining_indices]
+        active_request_indices = {
+            req_idx: orig_idx
+            for (req_idx, orig_idx) in enumerate(self.remaining_indices)
+        }
+        self.remaining_indices = []
+        if not request_entries:
+            return
+        try:
+            result_generator = self._gapic_fn(
+                timeout=next(self.timeout_generator),
+                entries=request_entries,
+                retry=None,
+            )
+            for result_list in result_generator:
+                for result in result_list.entries:
+                    orig_idx = active_request_indices[result.index]
+                    entry_error = core_exceptions.from_grpc_status(
+                        result.status.code,
+                        result.status.message,
+                        details=result.status.details,
+                    )
+                    if result.status.code != 0:
+                        self._handle_entry_error(orig_idx, entry_error)
+                    elif orig_idx in self.errors:
+                        del self.errors[orig_idx]
+                    del active_request_indices[result.index]
```

Comment:
active_request_indices is a map, not a list. It maps indexes between the original overall request, and the latest attempt (which may be for a small subset of the original mutations)

We should never encounter a result.index that isn't in active_request_indices, because they are both derived from remaining_indices